relay-server/src/services/project_cache.rs

```diff
@@ -455,16 +457,19 @@ impl ProjectSource {
             UpstreamProjectSourceService::new(config.clone(), upstream_relay).start();
 
         #[cfg(feature = "processing")]
-        let redis_maxconns = config.redis().map(|configs| {
-            let opts = match configs {
-                relay_config::RedisPoolConfigs::Unified((_, opts)) => opts,
-                relay_config::RedisPoolConfigs::Individual {
-                    project_configs: (_, opts),
-                    ..
-                } => opts,
-            };
-            opts.max_connections
-        });
+        let redis_maxconns = config
+            .redis()
+            .map(|configs| {
+                let config = match configs {
+                    relay_config::RedisPoolConfigs::Unified(config) => config,
+                    relay_config::RedisPoolConfigs::Individual {
+                        project_configs: config,
+                        ..
+                    } => config,
+                };
+                Self::compute_max_connections(config).unwrap_or(10)
+            })
+            .unwrap_or(10);
```

Comment:
I know this 10 was already in the previous version, but can we please make it a constant and document it? Especially now it's used in two places.

Reply:
Yup, I was considering of making it a constant. I have no idea about the rationale behind the value though, since it was there before i made my change. @Dav1dde do you have any ideas?

Reply:
¯\_(ツ)_/¯

Reply:
Great point in case about why constants should be documented. Maybe @gi0baro remembers? https://github.com/getsentry/relay/pull/3841

BTW it's totally fine to document that there is _no_ reason for the specific value of a constant (something like "we needed some upper bound but this parameter has not been fine-tuned").

Reply:
I think it was just a convenient number and I didn't think much about it in the review.